pkg/models/iam/im/im.go

```diff
@@ -154,24 +153,35 @@ func (im *defaultIMOperator) DescribeUser(username string) (*iamv1alpha2.User, e
 		klog.Error(err)
 		return nil, err
 	}
-
-	user := obj.(*iamv1alpha2.User)
-	out := user.DeepCopy()
-	// ensure encrypted password will not be output
-	out.Spec.EncryptedPassword = ""
-
-	return out, nil
+	return ensureNotControlledByFederationController(obj.(*iamv1alpha2.User)), nil
 }
 
 func (im *defaultIMOperator) DeleteUser(username string) error {
 	return im.ksClient.IamV1alpha2().Users().Delete(username, metav1.NewDeleteOptions(0))
 }
 
 func (im *defaultIMOperator) CreateUser(user *iamv1alpha2.User) (*iamv1alpha2.User, error) {
+	user = ensureNotControlledByFederationController(user)
 	user, err := im.ksClient.IamV1alpha2().Users().Create(user)
 	if err != nil {
 		klog.Error(err)
 		return nil, err
 	}
 	return user, nil
 }
+
+func ensurePasswordNotOutput(user *iamv1alpha2.User) *iamv1alpha2.User {
+	out := user.DeepCopy()
+	// ensure encrypted password will not be output
+	out.Spec.EncryptedPassword = ""
+	return out
+}
+
+func ensureNotControlledByFederationController(user *iamv1alpha2.User) *iamv1alpha2.User {
+	if user.Labels == nil {
+		user.Labels = make(map[string]string, 0)
+	}
+	// user entry in global scope should not controlled by federatedusers controller
+	user.Labels[constants.KubefedManagedLabel] = "false"
```

Comment:
If you don't want this label, just remove it, don't set it to false.

Reply:
This logic moved to the controller.